lisa/features/infiniband.py

```diff
@@ -334,3 +343,27 @@ def install_open_mpi(self) -> None:
         make = node.tools[Make]
         make.make("", cwd=openmpi_folder)
         make.make_install(cwd=openmpi_folder)
+
+    def install_ibm_mpi(self) -> None:
+        node = self._node
+        # Intall Open MPI
+        wget = node.tools[Wget]
+        script_path = wget.get(
+            "https://partnerpipelineshare.blob.core.windows.net/mpi/"
+            "platform_mpi-09.01.04.03r-ce.bin",
+            executable=True,
+        )
+        node.execute(
+            f"{script_path} -i silent",
+            sudo=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="Failed to install IBM MPI.",
+        )
+        node.execute(
+            f"export MPI_IB_PKEY={self.get_pkey()}; make -j $(nproc)",
```

Comment:
This is not the ideal way to do this. I am looking for a work around for the environment being wiped when running with sudo.

Reply:
I created a fix in another PR, will merge it tomorrow.